File: Algorithm/2217.py
# 로프
# https://www.acmicpc.net/problem/2217
import sys
input = sys.stdin.readline
n = int(input())
rope = []
for _ in range(n):
    rope.append(int(input()))
rope.sort()
ans = 0
for i in range(n):
    ans = max(ans,rope[i]*(n-i))
print(ans)

File: Algorithm_js/2217.js
/**로프
 * https://www.acmicpc.net/problem/2217 */
const input = require("fs")
  .readFileSync("example.txt")
  .toString()
  .split("\n")
  .map(Number);
// const input = require('fs').readFileSync('/dev/stdin').toString().split('\n').map(Number);
const n = input[0];
const ropes = input.slice(1).sort((a, b) => b - a);

let maxi = 0;
for (let i = 0; i < n; i++) {
  maxi = Math.max(maxi, ropes[i] * (i + 1));
}
console.log(maxi);
